### ISSUE TITLE
Mob spawning handler somehow adheres to _GUESTS_ group permissions

### ISSUE THREAD

**Issue by Robijnvogel** [ForgeEssentials/ForgeEssentials#1816]
A few months ago I set my server up according to the permissions guide on this Github's wiki.
I removed all permissions for the _GUESTS_ group and then added permissions for walking around and interacting with doors and trapdoors. I also set up a members group to "whitelist" people.

All went well until a few weeks ago when I updated +-66 mods on the server among which FE, added Hardcore Ender Expansion and More Planets and reset the server world. I also set up a second overworld with a different world generator type. ATG for the Overworld and BoP for DIM10
After this, no hostile mobs were spawning at night or in caves, regardless of the gamerule being true and the difficulty being normal.

First I thought that using the Multiword function directly caused this issue, because completely resetting my FEData caused mobs to spawn again, but when I re-uploaded my old permissions to the server (not the Multiworld stuff) it stopped spawning mobs again.

I then, by sheer desperation, decided to try and clear the "remove all permissions" rule from the _GUESTS_ group. This worked. I have come to the conclusion that somehow, since the FE update, my mob spawning handler has been assigned to the _GUESTS_ group somehow. When I list the users in _GUESTS_, however, it returns empty.

Right now I have found a workaround for the issue by allowing the _GUESTS_ group permission to  "fe.protection.mobspawning.*" I still thought I should file an issue, because if the mob spawn handler is in the _GUESTS_ group, who knows what else may not be working.

**Comment by olee** [comment]
Will need some debugging.
@luacs1998: It is still some time until I am back so I am counting on you.
You also could need some debugging training :smile:

**Comment by luacs1998** [comment]
If only I could attach a debugger to a FTB Ultimate server...

(note to self: that workflow needs improving)

**Comment by olee** [comment]
You don't need to check the modpack for that.
You should just investigate, if the guest group is included when checking spawn permissions.

**Comment by olee** [comment]
Ok I finally found time to check this and can confirm that (at least in the newest version) the `_GUESTS_` group is not included when world permissions (NPCs, fire spread, mob spawning, etc.) are checked.

The only thing I could imagine is that you included the `_GUESTS_` group from the `_ALL_` group (which is theoretically .... and practically) possible.

I didn't check it with the FTB Ultimate modpack yet, but even that should in no case influence any of this FE internal stuff (i.e. it would be pretty useless to try it with the modpack because it should give the same result).

**Comment by Robijnvogel** [comment]
Could More Planets and/or Hardcore Ender Expansion somehow have changed that? For as far as I know they do not add a custom mob spawner, but if such a spawning handler would be added as a fakeplayer by some mod, wouldn't it be thrown in the __Guest__ group by default?

**Comment by olee** [comment]
This is not the problem.
The permission check itself is only issued by FE - no other mod can influence this.

**Comment by Robijnvogel** [comment]
So I must somehow have included the __ALL__ group in the __GUESTS__ group?
I do not remember actively doing so. How can I check this?

**Comment by olee** [comment]
Try checking `./world/FEData/permissions/groups/_ALL_.txt` and see if you find a permission `fe.internal.group.parents` or `fe.internal.group.includes`

**Comment by Robijnvogel** [comment]
Only:
"fe.internal.group=true
fe.internal.group.priority=0
fe.remote.chat.send=false
some.test.permission=true"
in that file. And the comments of course, but those don't really do anything.

**Comment by Morketh** [comment]
Recently  i noticed only 2 different mobs spawning Slim in the over-world and Gasts in the Neather at first i though nothing of it however i began to realize the absence of every other hostile in the game. no Zombies, skeletons Creepers, Spiders, Enderman. Even other mods like Natura that spawn out random hostiles are not spawning Twilight forest mobs are not spawning either. this occurred just after a FE update. Im not really sure about the rest of the community but Im tired of my server drowning in slime :) not really sure if its related to this report (I have checked my permission nodes) but thought it a good place to add the info

**Comment by olee** [comment]
@Morketh try `/p test fe.protection.mobspawn.natural.type.hostile` and see if it's perhaps disabled.
I recommend you use that command from console and not as a player.

**Comment by Morketh** [comment]
command from console on a test server instance (im waiting on a New OS on my server provider so i cant test that one yet) 
```
[Server thread/INFO] [net.minecraft.server.MinecraftServer]: fe.protection.mobspawn.hostile.natural.type.hostile = true (not set)
```
all Mobs are spawning. Ill get the main server up and test it again the test environment only had FML + FE and everything is working within normal parameters. my only other guess is one of the other mods i updated might have caused it or the permission isnt set on the main server.

**Comment by Robijnvogel** [comment]
Could you link a list of your mods? If this is an issue we both have with an other mod, we may be able to find it this way.

**Comment by Morketh** [comment]
@Robijnvogel [Server List](https://gist.github.com/Morketh/d36d47b8219ad39f1e14)

**Comment by Robijnvogel** [comment]
It's not much, but these mods are not causing your problem if your cause is the same as mine:
-ABO
-Age Of Minecraft
-Buildcraft compat
-Magical Crops
-Mekanism etc.
-Modular Powersuits and Numina
-MyEssentials and MyPermissions
-MyTown
-Open CC sensors
-Pam's
-Project Red
Because I don't have these installed.

I wonder if Dynmap or Worldedit could be the culprit.

**Comment by Morketh** [comment]
Dynmap + World Edit both have permission nodes but as far as im aware they dont do anything with mob spawning. I was kinda leaning Towards My Town2 as this is a protection mod, however since you dont have that one installed im entirely at a loss as to what it could be. After a brand new map the mod spawns are back. I have the old server packed up on a spare drive ill pull that one out and run a few tests.

**Comment by Robijnvogel** [comment]
Have you done anything with the _GUESTS_ permission group? Like I described above, it is disallowing the _GUESTS_ group permission to fe.protection.mobspawning.*, that breaks the mobspawning in my case.

Before, I had disallowed that group all permissions, because new players automatically get put into this group.

**Comment by olee** [comment]
I was not able to replicate your issue.
Can you please zip and upload your permissions folder `world/FEdata/permissions`?

**Comment by Morketh** [comment]
@Robijnvogel I have not actually done anything with the _GUEST_ group however that is not to say that something else hasn't. 

Ill get an [upload on dropbox](https://www.dropbox.com/s/1kp977j4csqy09y/permissions.zip?dl=0) of the FE Perms so y'all can dig a bit deeper into those. On a fresh server i was unable to replicate the issue from scratch so it has to be something i changed on the other environment. At the moment server is failing to start due to another bug/error so im unable to provide a console readout of ```/p test fe.protection.mobspawn.natural.type.hostile```, however as soon as i get the new issue resolved i will post the results. [Issue unrelated to FE](https://github.com/dan200/ComputerCraft/issues/18) I feel like i spend more time repairing, tweaking and bug patching on my modpack then i do actually playing on it

**Comment by Robijnvogel** [comment]
I can find nothing weird on mobspawning permissions in that Dropbox, but I think only Olee or Luacs1998 can tell you for sure.
As for your other issue: probably Buildcraft Compat

**Comment by luacs1998** [comment]
Does this still happen with the permission fixes in the latest builds?

**Comment by Morketh** [comment]
I'm not sure I would have to dig out my server update everything load it up
and see

**Comment by luacs1998** [comment]
@pitchbright can you see if this is happening on your server with the latest FE builds?

**Comment by PitchBright** [comment]
I use a custom spawner, so I'm not sure if what I'm doing would be nearly the same as whatever everybody else is doing that is experiencing this issue (which I don't understand anyway). I've never seen mobspawning come up in my /p debug output.

**Comment by olee** [comment]
@PitchBright mob spawn permissions are hidden from `/p debug` - the same as some other permissions.
That is because it would spam the chat soo excessively, that you wouldn't be able to check anything else any more :smile:

**Comment by olee** [comment]
Ok - does it still happen for anyone that **guest permissions** affect mob spawning?
At least my tests show that this does not happen at all.
If so I would like to close this issue.

**Comment by luacs1998** [comment]
By all means, please do.

**Comment by Robijnvogel** [comment]
I can confirm that this is not an issue anymore

**Comment by olee** [comment]
@luacs1998 why did you reopen this if he said this is **not** an issue any more :smile:

**Comment by Robijnvogel** [comment]
XD good luck guys, I'll see you in the next issue.

lucacs, **don't ** reopen this again, ok? :tongue:
